tests/core/utilities/test_event_filter_builder.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+
+from hypothesis import (
+    given,
+    strategies as st,
+)
+
+from web3.utils.events import (
+    DataArgumentFilter,
+    TopicArgumentFilter,
+    normalize_topic_list,
+)
+
+
+@pytest.mark.parametrize(
+    "topic_list,expected",
+    (
+        (
+            ("0x1", "0x2", ["0x3"], None, "0x4", None, None, None),
+            ("0x1", "0x2", "0x3", None, "0x4")
+        ),
+        (
+            (None, ["0x2", "0x2a"], "0x3", None, "0x4", None, [None], None),
+            (None, ["0x2", "0x2a"], "0x3", None, "0x4")
+        ),
+        (
+            (None, None, [None]),
+            tuple()
+        )
+    )
+)
+def test_normalize_topic_list(topic_list, expected):
+    assert normalize_topic_list(topic_list) == expected
+
+
+@given(st.text())
+def test_match_single_string_type_properties_data_arg(value):
+    ea = DataArgumentFilter(arg_type="string")
+    ea.match_single(value)
+
+
+@given(st.text())
+def test_match_single_string_type_properties_topic_arg(value):
+    ea = TopicArgumentFilter(arg_type="string")
+    ea.match_single(value)
```

Comment:
Should assert the encoded topic value is correct.

Reply:
What does `ea` stand for?